sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java

```diff
@@ -72,40 +68,55 @@ public CxxCoverageSensor(Settings settings, FileSystem fs, ProjectReactor reacto
     parsers.add(new VisualStudioParser());
   }
 
+  @Override
+  public boolean shouldExecuteOnProject(Project project) {
+    return fs.hasFiles(fs.predicates().hasLanguage(CxxLanguage.KEY));
+  }
+  
   /**
    * {@inheritDoc}
    */
   @Override
   public void analyse(Project project, SensorContext context) {
 
-    CxxUtils.LOG.debug("Parsing coverage reports");
-    List<File> reports = getReports(conf, reactor.getRoot().getBaseDir().getAbsolutePath(), REPORT_PATH_KEY, DEFAULT_REPORT_PATH);
-    if (reports.isEmpty()) {
-        reports = getReports(conf, fs.baseDir().getPath(), REPORT_PATH_KEY, DEFAULT_REPORT_PATH);
+    Map<String, CoverageMeasuresBuilder> coverageMeasures = null;
```

Comment:
All other sensor with reports are looking first if there is the report key in the sonar-project.properties if not the sensor is not executed. Coverage is different for two reasons:
1) There are three sonar-project.properties keys and not only one.
2) If no report is available there is still the 'sonar.cxx.coverage.forceZeroCoverage'

Inside of analyse I changed things. Each key is checked and if not available nothing is will be done. If also sonar.cxx.coverage.forceZeroCoverage is false then analyse is doing nothing and should be fast.